lib/inat_inferrer.py

```diff
@@ -556,17 +556,32 @@ def limit_leaf_scores_that_include_humans(self, leaf_scores):
 
     @staticmethod
     def prepare_image_for_inference(file_path):
-        mime_type = magic.from_file(file_path, mime=True)
-        # attempt to convert non jpegs
-        if mime_type != "image/jpeg":
-            im = Image.open(file_path)
-            image = im.convert("RGB")
-        else:
-            image = tf.io.read_file(file_path)
-            image = tf.image.decode_jpeg(image, channels=3)
+        image = Image.open(file_path)
+        if image.mode != "RGB":
+            image = image.convert("RGB")
         image = tf.image.convert_image_dtype(image, tf.float32)
-        image = tf.image.resize_with_crop_or_pad(
-            image, 299, 299
+
+        eventual_size = 299
+        central_crop_factor = 0.875
+        resize_min_dimension = eventual_size/central_crop_factor
+
+        height, width = image.shape[0], image.shape[1]
+        resize_ratio = np.array([height, width]).min() / resize_min_dimension
```

Comment:
could just use python `min(x, y)` or `min([x, y])` instead of needing numpy if you want